Source/Core/VideoCommon/PixelShaderGen.cpp

```diff
@@ -870,10 +870,28 @@ ShaderCode GeneratePixelShaderCode(APIType ApiType, const ShaderHostConfig& host
     }
     else
     {
-      out.Write("\tif(bbox_left > int(rawpos.x)) atomicMin(bbox_left, int(rawpos.x));\n"
-                "\tif(bbox_right < int(rawpos.x)) atomicMax(bbox_right, int(rawpos.x));\n"
-                "\tif(bbox_top > int(rawpos.y)) atomicMin(bbox_top, int(rawpos.y));\n"
-                "\tif(bbox_bottom < int(rawpos.y)) atomicMax(bbox_bottom, int(rawpos.y));\n");
+      out.Write(R"(
+  int2 minpos = int2(rawpos.xy);
+  int2 maxpos = int2(rawpos.xy);
+
+#ifdef SUPPORTS_WARP_REDUCTION
+  WARP_MIN(minpos);
+  WARP_MAX(maxpos);
+  if (IS_FIRST_ACTIVE_WARP)
```

Comment:
It's possible that this new condition will prevent the loads for the bbox_* values from being scheduled near the beginning of the shader, so it may actually end up being slower. Maybe help the compiler a bit by moving them there manually.